comment_freeze test case fails

Looks like #536/#558 break some old tests:
```
    Test case "operation_time_tests/comment_freeze" was aborted with:
      18 assertions out of 20 passed
      2 assertions out of 20 failed
```

```
golos/tests/tests/operation_time_tests.cpp:2473: Entering test case "comment_freeze"
golos/tests/tests/operation_time_tests.cpp:2585: fatal error: in "operation_time_tests/comment_freeze": exception fc::exception expected but not raised
unknown location:0: fatal error: in "operation_time_tests/comment_freeze": unknown type
```

--- a/libraries/chain/block_log.cpp
+++ b/libraries/chain/block_log.cpp
@@ -76,7 +76,7 @@ namespace golos { namespace chain {
 
                 const auto* ptr = block_mapped_file.data() + pos;
                 const auto available_size = file_size - pos;
-                const auto max_block_size = std::min(available_size, std::size_t(STEEMIT_MAX_BLOCK_SIZE));
+                const auto max_block_size = std::min<std::size_t>(available_size, STEEMIT_MAX_BLOCK_SIZE);
 
                 fc::datastream<const char*> ds(ptr, max_block_size);
                 fc::raw::unpack(ds, block);

--- a/libraries/chain/steem_evaluator.cpp
+++ b/libraries/chain/steem_evaluator.cpp
@@ -2230,6 +2230,7 @@ namespace golos { namespace chain {
             auto min_delegation = median_fee * GOLOS_MIN_DELEGATION_MULTIPLIER * v_share_price;
             auto min_update = median_fee * v_share_price;
 
+            auto now = _db.head_block_time();
             auto delta = delegation ?
                 op.vesting_shares - delegation->vesting_shares :
                 op.vesting_shares;
@@ -2243,14 +2244,20 @@ namespace golos { namespace chain {
 #endif
 
             if (increasing) {
+                auto delegated = delegator.delegated_vesting_shares;
                 FC_ASSERT(delegator.available_vesting_shares(true) >= delta,
                     "Account does not have enough vesting shares to delegate.",
                     ("available", delegator.available_vesting_shares(true))
-                    ("delta", delta)
-                    ("vesting_shares", delegator.vesting_shares)
-                    ("delegated_vesting_shares", delegator.delegated_vesting_shares)
-                    ("to_withdraw", delegator.to_withdraw)
-                    ("withdrawn", delegator.withdrawn));
+                    ("delta", delta)("vesting_shares", delegator.vesting_shares)("delegated", delegated)
+                    ("to_withdraw", delegator.to_withdraw)("withdrawn", delegator.withdrawn));
+                auto elapsed_seconds = (now - delegator.last_vote_time).to_seconds();
+                auto regenerated_power = (STEEMIT_100_PERCENT * elapsed_seconds) / STEEMIT_VOTE_REGENERATION_SECONDS;
+                auto current_power = std::min<int64_t>(delegator.voting_power + regenerated_power, STEEMIT_100_PERCENT);
+                auto max_allowed = delegator.vesting_shares * current_power / STEEMIT_100_PERCENT;
+                FC_ASSERT(delegated + delta <= max_allowed,
+                    "Account allowed to delegate a maximum of ${v} with current voting power = ${p}",
+                    ("v",max_allowed)("p",current_power)("delegated",delegated)("delta",delta));
+
                 if (!delegation) {
                     FC_ASSERT(op.vesting_shares >= min_delegation,
                         "Account must delegate a minimum of ${v}", ("v",min_delegation)("vesting_shares",op.vesting_shares));
@@ -2258,7 +2265,7 @@ namespace golos { namespace chain {
                         o.delegator = op.delegator;
                         o.delegatee = op.delegatee;
                         o.vesting_shares = op.vesting_shares;
-                        o.min_delegation_time = _db.head_block_time();
+                        o.min_delegation_time = now;
                     });
                 }
                 _db.modify(delegator, [&](account_object& a) {
@@ -2271,7 +2278,7 @@ namespace golos { namespace chain {
                 _db.create<vesting_delegation_expiration_object>([&](vesting_delegation_expiration_object& o) {
                     o.delegator = op.delegator;
                     o.vesting_shares = -delta;
-                    o.expiration = std::max(_db.head_block_time() + STEEMIT_CASHOUT_WINDOW_SECONDS, delegation->min_delegation_time);
+                    o.expiration = std::max(now + STEEMIT_CASHOUT_WINDOW_SECONDS, delegation->min_delegation_time);
                 });
             }
 

--- a/libraries/protocol/include/golos/protocol/asset.hpp
+++ b/libraries/protocol/include/golos/protocol/asset.hpp
@@ -47,8 +47,15 @@ namespace golos { namespace protocol {
                 amount *= x;
                 return *this;
             }
+            asset& operator/=(int x) {
+                amount /= x;
+                return *this;
+            }
             asset operator*(int x) const {
                 return asset(amount * x, symbol);
+            }
+            asset operator/(int x) const {
+                return asset(amount / x, symbol);
             }
 
             asset operator-() const {

--- a/tests/tests/operation_tests.cpp
+++ b/tests/tests/operation_tests.cpp
@@ -6469,6 +6469,7 @@ BOOST_FIXTURE_TEST_SUITE(operation_tests, clean_database_fixture)
 
             generate_block();
             vest("alice", ASSET_GOLOS(10000));
+            vest("bob", ASSET_GOLOS(1000));
             generate_block();
             db_plugin->debug_update([=](database& db) {
                 db.modify(db.get_witness_schedule_object(), [&](witness_schedule_object& w) {
@@ -6532,6 +6533,16 @@ BOOST_FIXTURE_TEST_SUITE(operation_tests, clean_database_fixture)
                 (old_voting_power - bob_acc.voting_power) / STEEMIT_100_PERCENT;
             BOOST_REQUIRE(rshares == itr->rshares);
             BOOST_REQUIRE(rshares == alice_comment.net_rshares.value);
+
+            BOOST_TEST_MESSAGE("--- Test that delegation limited by current voting power");
+            auto max_allowed = bob_acc.vesting_shares * bob_acc.voting_power / STEEMIT_100_PERCENT;
+            op.delegator = "bob";
+            op.delegatee = "alice";
+            op.vesting_shares = asset(max_allowed.amount + 1, VESTS_SYMBOL);
+            sign_tx_with_ops(tx, bob_private_key, op);
+            STEEMIT_REQUIRE_THROW(db->push_transaction(tx), fc::assert_exception);
+            op.vesting_shares = max_allowed;
+            push_tx_with_ops(tx, bob_private_key, op);
 
             generate_block();
             ACTORS((sam)(dave))

--- a/tests/tests/operation_tests.cpp
+++ b/tests/tests/operation_tests.cpp
@@ -6681,11 +6681,14 @@ BOOST_FIXTURE_TEST_SUITE(operation_tests, clean_database_fixture)
             generate_blocks(10);
 
             auto acc = db->get_account("alice");
+#ifndef IS_LOW_MEM
             auto meta = db->get<account_metadata_object, by_account>("alice");
             BOOST_REQUIRE(meta.account == "alice");
             BOOST_REQUIRE(meta.json_metadata == json);
+#endif
             BOOST_REQUIRE(acc.last_account_update == now);
 
+#ifndef IS_LOW_MEM
             BOOST_TEST_MESSAGE("--- Test existance of account_metadata_object after account_create");
             ACTOR(bob);                                             // create_account with json_metadata = ""
             meta = db->get<account_metadata_object, by_account>("bob");
@@ -6704,12 +6707,13 @@ BOOST_FIXTURE_TEST_SUITE(operation_tests, clean_database_fixture)
             cr.owner = authority(1, priv_key.get_public_key(), 1);
             cr.active = authority(1, priv_key.get_public_key(), 1);
             cr.memo_key = priv_key.get_public_key();
-            // op.json_metadata = "";                               // don't set
+            // cr.json_metadata = "";                               // don't set
             push_tx_with_ops(tx, bob_private_key, cr);
 
             meta = db->get<account_metadata_object, by_account>("sam");
             BOOST_REQUIRE(meta.account == "sam");
             BOOST_REQUIRE(meta.json_metadata == "");
+#endif
             validate_database();
         }
         FC_LOG_AND_RETHROW()

--- a/tests/tests/operation_time_tests.cpp
+++ b/tests/tests/operation_time_tests.cpp
@@ -2582,7 +2582,8 @@ BOOST_FIXTURE_TEST_SUITE(operation_time_tests, clean_database_fixture)
 
             tx.operations.push_back(comment);
             tx.sign(alice_private_key, db->get_chain_id());
-            STEEMIT_REQUIRE_THROW(db->push_transaction(tx, 0), fc::exception);
+            // comments do not expire now
+            db->push_transaction(tx, 0);
         }
         FC_LOG_AND_RETHROW()
     }

